src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/controller/PostController.java

```diff
@@ -30,55 +28,53 @@ public class PostController {
 
     private final PostService postService;
     private final PostMapper postMapper;
+    private final PostRequestValidator postRequestValidator;
+    private final BindingResultService bindingResultService;
 
     @PostMapping
-    public ResponseEntity<?> createPost(@RequestBody @Valid PostRequestDTO postRequestDTO,
+    public ResponseEntity<?> createPost(@RequestBody @Valid PostRequest postRequest,
                                         BindingResult bindingResult) {
 
-        if (bindingResult.hasErrors()) {
-            return handleErrorsFromBindingResult(bindingResult);
-        }
+        postRequestValidator.validate(postRequest, bindingResult);
```

Comment:
We discussed this matter earlier. I like the idea of validating it, but I don't like the idea of making an additional database request for each new request. 
Also, I'm not in favor of using the repository outside of the service. 

I'm not sure why open transactions are enabled in the controller by default, but it can't be appropriate to make database calls from any place in the request lifecycle. 

My suggestions:
    Controller Layer: Perform basic syntactical validations using annotations.
    Service Layer: Conduct business logic and domain-specific validations, including those requiring database access.
    
We need to weigh all the pros and cons of such a decision again, maybe @oleksandr-jr  can help.

Reply:
Yes, there are a lot of writing options, it's better to know what the best practices are here so that it looks concise and reads well @oleksandr-jr